### PR TITLE
Improve Contributing

### PR DESCRIPTION
**Pull request by scott-w**
### Proposed changes
 - Change the branch choice docs
 - Added approval process
 - Put the PR process into a top-level header

Link to the issue: #3185

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -56,72 +51,80 @@ Please refrain from giving code examples in altJS languages like CoffeeScript,
...
 
-We'll even accept a failing test pasted into the ticket description instead of a
-PR. That would at least get us started on creating the failing test in the code.
+Once two Marionette.js members approve the pull request, we will then merge it
+into the base branch.
+
+Please remember that Marionette is a community-maintained project and, as such,
+many of us are working on this in our spare time. If we haven't commented on
+your pull request, please be patient. We may be available on our Gitter channel
+to discuss further.
```
Comment (denar90) [inline CONTRIBUTING.md]: Good remark 👍

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -3,29 +3,24 @@ the process of getting changes merged in and released.
 
 1. [Setting up Marionette locally](#setting-up-marionette-locally)
 2. [Reporting a bug](#reporting-a-bug)
-  1. [When you have a patch](#when-you-have-a-patch)
-  2. [When you don't have a bug fix](#when-you-dont-have-a-bug-fix)
-3. [Running Tests](#running-tests)
-4. [Updating docs](#updating-docs)
+3. [Submitting patches and fixes](#submitting-patches-and-fixes)
+4. [Running Tests](#running-tests)
```
Comment (denar90) [inline CONTRIBUTING.md]: Some titles are missed:
 - `When you don't have a bug fix`
 - `Determining your branch`
 - `Submitting a Great Patch`
 - `Solving Issues`
 - `Coding Guidelines`
 - `How we Approve Pull Requests`

Maybe the were not added in purpose.

Comment (scott-w) [inline CONTRIBUTING.md]: I wanted to stick to the top-level headers here so the reader could quickly find the thing they're currently working on e.g. they're submitting a pull request.

That said, I can definitely see the arguments for including them as someone might be thinking "what do I do for docs!"

Comment (denar90) [inline CONTRIBUTING.md]: Got your point.

**Pull request merged**